packages/flutter/lib/src/services/image_stream.dart

```diff
@@ -365,6 +365,9 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
 
   void _handleCodecReady(ui.Codec codec){
     _codec = codec;
+    if (_codec == null)
+      throw new Exception('Codec future completed with null');
```

Comment:
i would just `assert(_codec != null)`, to make it clearer that null is not ok. Also, document that in the API docs.